pkg/services/featuremgmt/registry.go

```diff
@@ -1221,6 +1221,14 @@ var (
 			HideFromDocs:      true,
 			HideFromAdminPage: true,
 		},
+		{
+			Name:            "alertingUpgradeDryrunOnStart",
+			Description:     "Dry-run Unified Alerting upgrade on startup",
```

Comment:
Since we advertise this flag in the docs, wdyt about more elaborate description that mentions that it runs on start up and logs all problems.

Reply:
@armandgrillet asked for a copy review. 

1) Suggestion for the feature flag description which could be used in different places in this PR. Feel free to adjust it or change completely.  

> Enable the dry-run upgrade for Legacy alerts on startup to transition them to the Unified Alert format. Legacy alerts will not be supported in v11. This upgrade logs the results and is reversible.

2) We should probably include a `Technical note` on the `Upgrade to Grafana v10.4` documentation, similar to this [example](https://grafana.com/docs/grafana/latest/upgrade-guide/upgrade-v9.5/#migrate-your-api-keys-to-service-accounts).

Reply:
1. How does this sound: `When activated in legacy alerting mode, this feature flag initiates a dry-run of the Unified Alerting upgrade during each startup. It logs any issues detected without implementing any actual changes.`
2. Good idea!

Reply:
👍